### ISSUE TITLE
Generator: Implement Composition as abstract class to inherit redundant fields for other Compositions 

### ISSUE THREAD

**Issue by SevKohler** [ehrbase/openEHR_SDK#94]
## Background ##

Currently many of the generated Compositions contain duplicated fields like Language, these Attributes could be contained by a root class Composition, which makes specific Compositions smaller and allows easier handling of Compositions when programming. Especially when in general Compositions are to be referenced.

## Acceptance criteria ##

Implement (abstract) class Composition that contains all Attributes that are standard to every Composition. 

## Success criteria ##

- [ ] Review / Merge request approved (P2P session)
- [ ] Unit tests passed
- [ ] Updated documentation (Javadoc and Sphinx)
- [ ] Acceptance criteria fulfilled
- [ ] Build without errors
- [ ] Release notes prepared
- [ ] Runtime warnings

**Comment by ppazos** [comment]
I think we can have an AbstractComposition in the shared package that all generated Composition DTOs inherit from.

**Comment by subigre** [comment]
+1 I would appreciate if we might have this abstract Composition class. 
This could help to implement the FHIR Bridge and the openEHR components for Camel having something similar to the `MessageConverter` provided by Spring for JMS.

**Comment by subigre** [comment]
One additional comment, it might be interesting to update the CompositionEndpoint:

```
public interface CompositionEndpoint {

    <T extends Composition> T mergeCompositionEntity(T entity);

    <T extends Composition> Optional<T> find(UUID compositionId, Class<T> clazz);
}
```

**Comment by SevKohler** [comment]
I think an AbstractComposition is better, this is also because of the semantics itself and the Mandatory Fields we always have to set,:
Example:
```Java
observation.setOriginValue(effectiveDateTime);
observation.setTimeValue(effectiveDateTime);
observation.setLanguage(Language.DE);
```
Using an abstract class for that is better in design and use in this case. 
Also implementing here to get the language from the Template (//FIXME comment) would eradicate a lot of duplicates.
An Interface is to be better to be used when classes have the same methods but differ in their execution (and have no hierarchical relations), we have here several cases where we need the same method for subclasses.  About the Converter we need to discuss this further.

**Comment by stefanspiska** [comment]
Fixed in #153 which will become release 1.0.0
see
https://github.com/ehrbase/openEHR_SDK/blob/release/v1.0.0/CHANGELOG.md for changes
https://github.com/ehrbase/openEHR_SDK/blob/release/v1.0.0/UPDATING.md for how to update
